buildspec.pypi.yml

```diff
@@ -0,0 +1,41 @@
+version: 0.2
+
+env:
+  shell: bash
+  parameter-store:
+    PYPI_TOKEN: "pypi-token"
+
+phases:
+  install:
+    commands:
+      - echo Setting local Python versions
+      - pyenv versions | awk 'match($0, /\d\.\d+\.\d+/) { print substr($0, RSTART, RLENGTH) }' > .python-version
+      - echo Installing dependencies
+      - pip install poetry
+      - poetry install
+
+  build:
+    commands:
+      - python --version
+      - echo Running check-cloudformation
+      - poetry run ./copilot_helper.py check-cloudformation
+      - echo Running tests
+      - poetry run tox
+      - |
+        if ! python utils/check_pypi.py
+        then
+          echo Ensure we are using python 3.11
+          pyenv local 3.11
+          echo Building python package
+          pip install poetry
+          poetry install
```

Comment:
I wonder if we could do a `pyenv global 3.11` in the `install` phase to avoid having to install `poetry` and all the dependencies again.

Reply:
Resolved in https://github.com/uktrade/copilot-tools/pull/139/commits/22ab4bf423dcd58c1b3e639e6fceaed250e85267.